pyproject.toml

```diff
@@ -138,8 +138,18 @@ ignore = [
   "UP007",   # Prefer Optional[], Union[] over | due to torch jit scripting
   "UP006",   # Prefer List[], over list due to torch jit scripting
   "UP035",   # Ignore deprecated typing because of jit scripting
-  'D100', 'D101', 'D102','D103', 'D104', 'D105', 'D107', 'D203', 'D204', 'D205', 'D213', 'D400', 'D401', 'D404', 'D406', 'D407', 'D415', 'D417'# docs -- TODO: unconment some of these rules..
-]
+  'D100',    # Allow Undocumented public module
+  'D101',    # TODO: Undocumented public class
+  'D102',    # TODO: Undocumented public method
+  'D103',    # TODO: Undocumented public function
+  'D104',    # TODO: Undocumented public package
+  'D105',    # Allow Undocumented magic method
+  'D107',    # TODO: Undocumented public init
+  'D417',    # TODO: Undocumented params
+ ]
+
+[tool.ruff.lint.pydocstyle]
+convention = "google"
```

Comment:
could you share some examples of why use Google here instead of numpy? 

I have been using numpy with ruff into other projects without issues

Reply:
I have no strong opinions about Numpy style with ruff globally (I also prefer numpy docstyle generally). For this PR I think there were two arguments for it:

1. The current format is much closer to google style (args instead of parameters, no underlines etc), so reducing the scope of the PR to be enabling/fixing the rules rather than converting docstring style.
  - When I set the numpy doc string style it didn't auto-convert Args: to Parameters with the underlines. I was going to do a find replace to resolve that but fell down the rabbit hole of non-compliant style details (some people have used `Inputs` or `Arg` or other things, so hard to say that the style after the refactor would be consistent without manually skimming everything)

2. https://github.com/astral-sh/ruff/issues/2310 D417 doesn't work for numpy style AFAICT


Example for 2:
```python
from typing import Any


# No error
def google_all_args_documented(arg_1: Any, arg_2: Any):
    """Calculate something.

    Args:
        arg_1: first arg
        arg_2: second arg

    Returns:
        Nothing.
    """
    pass


# D417
def google_some_args_documented(arg_1: Any, arg_2: Any):  # noqa: D417
    """Calculate something.

    Args:
        arg_1: first arg

    Returns:
        Nothing.
    """
    pass


# No error
def numpy_some_args_documented(arg_1: Any, arg_2: Any):
    """Calculate something.

    Parameters
    ----------
        arg_1: first arg

    Returns:
    -------
        Nothing.
    """
    pass
```

Reply:
i see, tks